bin/publish.ml

```diff
@@ -31,6 +30,21 @@ let publish_doc ~dry_run ~yes pkg_names pkg =
   gen_doc ~dry_run ~force dir pkg_names >>= fun docdir ->
   Delegate.publish_doc ~dry_run ~yes pkg ~msg ~docdir
 
+(* If the `doc` field of the opam file is not set we do not generate nor
+   publish the documentation, except when using a delegate. *)
+let publish_doc ~dry_run ~yes pkg_names pkg =
+  App_log.status (fun l -> l "Publishing documentation");
+  match Pkg.doc_uri pkg with
+  | Error _ | Ok "" -> (
+      match Pkg.delegate pkg with
+      | Ok (Some _) -> publish_doc ~dry_run ~yes pkg_names pkg
+      | Error _ | Ok None ->
+          Pkg.name pkg >>= fun name ->
+          App_log.status (fun l -> l "No doc field found for package %s" name);
```

Comment:
Regarding the user logs, what would you think about:
- if there's is a doc field, we log `Publishing documentation` and go on with it
- if there's none, we log `Skipping documentation publication as there is no 'doc' field in xxx.opam`

I think it's a bit better than saying that we're going to publish the doc only to say we're not going to.
Might be nice to point to the opam file as well, to avoid confusion about where the doc field is.
I think it seems obvious to us but to new comers or to people that aren't well aware of dune-release's behaviour it might not.

Reply:
Maybe `Skipping documentation publication for package x: no 'doc field in x.opam` is a better message than what I suggested above!

What do you think?